File: frontend/frontend_idea_app/lib/models/Message.dart
class Message {
  String text, timestamp, whoSent;
  Map<String, bool> visibility;

  Message(this.text, this.timestamp, this.visibility, this.whoSent);

  Message.fromJson(Map<String, dynamic> json)
      : timestamp = json['timestamp'],
        whoSent = json['whoSent'],
        text = json['text'];

  Map<String, dynamic> toJson() => {
        "timestamp": "'" + timestamp + "'",
        "text": "'" + text + "'",
        "whoSent": "'" + whoSent + "'"
      };
}

File: backend/ideappservice/src/main/java/com/ideapp/ideappservice/model/Message.java
package com.ideapp.ideappservice.model;

import java.util.HashMap;
import java.util.Map;

public class Message {
    String text;
    String timestamp;
    Map<String, Boolean> visibility;
    String whoSent;
    public Message(){}

    public Message(String text, String timestamp, Map<String, Boolean> visibility,String whoSent) {
        this.text = text;
        this.timestamp = timestamp;
        this.visibility = visibility;
        this.whoSent = whoSent;
    }

    public String getWhoSent() {
        return whoSent;
    }

    public void setWhoSent(String whoSent) {
        this.whoSent = whoSent;
    }

    public String getText() {
        return text;
    }

    public void setText(String text) {
        this.text = text;
    }

    public String getTimestamp() {
        return timestamp;
    }

    public void setTimestamp(String timestamp) {
        this.timestamp = timestamp;
    }

    public Map<String, Boolean> getVisibility() {
        return visibility;
    }

    public void setVisibility(HashMap<String, Boolean> visibility) {
        this.visibility = visibility;
    }
}
